doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,524 @@
+     terminated with a period and optionally decorated with
+     :term:`attributes <attribute>`.
+
+     .. insertprodn document sentence
+
+     .. prodn::
+        document ::= {* @sentence }
+        sentence ::= {? @attributes } @command .
+        | {? @attributes } {? @num : } @query_command .
+        | {? @attributes } {? @toplevel_selector } @ltac_expr {| . | ... }
+        | @control_command
+
+     :n:`@ltac_expr` syntax supports both simple and compound
+     :term:`tactics <tactic>`.  For example: ``split`` is a simple
+     tactic while ``split; auto`` combines two simple tactics.
+
+   command
+
+     A :production:`command` can be used to modify the state of a |Coq|
+     document, for instance by declaring a new object, or to get
+     information about the current state.
+
+     By convention, command names begin with uppercase letters.
+     Commands appear in the HTML documentation in blue or gray boxes
+     after the label "Command".  In the pdf, they appear after the
+     boldface label "Command:".  Commands are listed in the
+     :ref:`command_index`.  Example:
+
+     .. cmd:: Comments {* @string }
+
+        This command prints "Comments ok" and does not change anything
+        to the state of the document.
+
+   tactic
+
+     Tactics specify how to transform the current proof state as a
+     step in creating a proof.  They are syntactically valid only when
+     |Coq| is in proof mode, such as after a :cmd:`Theorem` command
+     and before any subsequent proof-terminating command such as
+     :cmd:`Qed`.  See :ref:`proofhandling` for more on proof mode.
+
+     By convention, tactic names begin with lowercase letters.  Tactic
+     appear in the HTML documentation in blue or gray boxes after the
+     label "Tactic".  In the pdf, they appear after the boldface label
+     "Tactic:".  Tactics are listed in the :ref:`tactic_index`.
+
+Settings
+--------
+
+This section presents the mechanisms for changing the behavior of
+|Coq|.  First, we describe the :term:`attribute` mechanism, which is
+used to modify the behavior of a single :term:`sentence`.  Next, we
+describe the :term:`flag`, :term:`option` and :term:`table` mechanism,
+which is used to modify the behavior of |Coq| more globally in a
+document or project.
+
+.. _attributes:
+
+Attributes
+~~~~~~~~~~
+
+An :gdef:`attribute` modifies the behavior of a single sentence.
+Syntactically, most commands and tactics can be decorated with
+attributes (cf. :n:`@sentence`), but attributes not supported by the
+command or tactic will trigger :warn:`This command does not support
+this attribute`.
+
+.. insertprodn attributes legacy_attr
+
+.. prodn::
+   attributes ::= {* #[ {*, @attribute } ] } {* @legacy_attr }
+   attribute ::= @ident {? @attr_value }
+   attr_value ::= = @string
+   | ( {*, @attribute } )
+   legacy_attr ::= {| Local | Global }
+   | {| Polymorphic | Monomorphic }
+   | {| Cumulative | NonCumulative }
+   | Private
+   | Program
```

Comment:
Doesn't this give a too strong importance to legacy attributes? Wouldn't it be clearer to move them to their own section (doomed to be removed at some time).

Maybe the time is ripe also to apply a script which consistently turns all legacy-syntax attributes into `#[ ]` syntax in existing files??

Reply:
To be honest, they still have quite a strong importance. Given that 8.12 will be the first version of Coq where attributes are really put forward in the documentation, and given how frequently legacy attributes are used in practice (including in the documentation), it seems important to me not to hide them under the carpet.

Legacy attributes may be doomed to eventually disappear, but they do not currently create any particular trouble (except `Export` which couldn't be turned into a proper legacy attribute).